arch/x86_64/src/cmake/Toolchain.cmake

```diff
@@ -83,6 +83,12 @@ if(CONFIG_DEBUG_SYMBOLS)
   add_compile_options(${CONFIG_DEBUG_SYMBOLS_LEVEL})
 endif()
 
+if(CONFIG_HOST_LINUX)
+  if(CONFIG_ARCH_HAVE_FORK)
```

Comment:
`  if(CONFIG_ARCH_HAVE_FORK)` doesn't make sense in x86_64 port context, it is always set to `y`

Reply:
Since the warning is caused by linking fork.o:

So I added the macro as source file did:

vi arch/x86_64/src/common/Make.defs +28
ifeq ($(CONFIG_ARCH_HAVE_FORK),y)
CMN_CSRCS += x86_64_fork.c
CMN_ASRCS += fork.S
endif

Reply:
condition in `arch/x86_64/src/common/Make.defs` also doesn't make sense, this option is set permanently here:

https://github.com/apache/nuttx/blob/b28f87e3f0cd4ec901b2a716fea15378cb4cfd09/arch/Kconfig#L147